vit-servicing-station-tests/src/tests/snapshot.rs

```diff
@@ -0,0 +1,179 @@
+    for (idx, entry) in first_snapshot.content.iter().enumerate() {
+        assert!(
+            rest_client
+                .voting_power(&first_snapshot.tag, &entry.voting_key.to_hex())
+                .unwrap()
+                .is_empty(),
+            "expected empty data for entry idx: {}",
+            idx
+        );
+    }
+    for (idx, entry) in second_snapshot.content.iter().enumerate() {
+        let voting_power = VotingPower::from(entry.clone());
+        assert_eq!(
+            vec![voting_power],
+            rest_client
+                .voting_power(&second_snapshot.tag, &entry.voting_key.to_hex())
+                .unwrap(),
+            "expected empty data for entry idx: {}",
+            idx
+        );
+    }
+}
+
+#[test]
+pub fn import_snapshots_with_different_tags() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client = server.rest_client_with_token(&data.token_hash());
+
+    let first_snapshot = Snapshot::default();
+
+    rest_client.put_snapshot(&first_snapshot).unwrap();
+
+    let second_snapshot = SnapshotUpdater::from(first_snapshot.clone())
+        .update_tag("fund9")
+        .build();
+
+    rest_client.put_snapshot(&second_snapshot).unwrap();
+
+    for (idx, entry) in first_snapshot.content.iter().enumerate() {
+        let voting_power = VotingPower::from(entry.clone());
+        assert_eq!(
+            vec![voting_power.clone()],
+            rest_client
+                .voting_power(&first_snapshot.tag, &entry.voting_key.to_hex())
+                .unwrap(),
+            "wrong data for entry idx: {}",
+            idx
+        );
+        assert_eq!(
+            vec![voting_power],
+            rest_client
+                .voting_power(&second_snapshot.tag, &entry.voting_key.to_hex())
+                .unwrap(),
+            "wrong data for entry idx: {}",
+            idx
+        );
+    }
+}
+
+#[test]
+pub fn import_malformed_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client: RawRestClient = server.rest_client_with_token(&data.token_hash()).into();
+
+    let snapshot = Snapshot::default();
+    let mut content = serde_json::to_string(&snapshot.content).unwrap();
+    content.pop();
+    assert!(rest_client
+        .put_snapshot(&snapshot.tag, content)
+        .unwrap()
+        .status()
+        .is_client_error());
+}
+#[test]
+pub fn import_big_snapshot() {
+    let temp_dir = TempDir::new().unwrap();
+    let (server, data) = quick_start(&temp_dir).unwrap();
+    let rest_client = server.rest_client_with_token(&data.token_hash());
+
+    let snapshot = SnapshotBuilder::default()
+        .with_tag("big")
+        .with_entries_count(100_000)
+        .with_groups(vec![
+            "daily".to_string(),
+            "fund8".to_string(),
+            "fund9".to_string(),
+            "fund10".to_string(),
```

Comment:
these names don't really look like what I would expect as `groups`, but it's mostly irrelevant